cmd/repo-init/main.go

```diff
@@ -462,25 +462,46 @@ func fetchOrDefaultWithPrompt(msg, def string) string {
 	return ""
 }
 
+type RepoProwConfig struct {
+	Tide TideRepoProwConfig `json:"tide,omitempty"`
+}
+
+type TideRepoProwConfig struct {
+	Queries prowconfig.TideQueries `json:"queries,omitempty"`
+}
+
 func updateProwConfig(config initConfig, releaseRepo string) error {
 	configPath := path.Join(releaseRepo, ciopconfig.ConfigInRepoPath)
+	configDir := path.Dir(configPath)
+	tmpDir, err := ioutil.TempDir("", "repo-init")
+	if err != nil {
+		return fmt.Errorf("failed to create temp dir: %w", err)
+	}
+	defer os.Remove(tmpDir)
+
+	targetOrgRepo := prowconfig.OrgRepo{Org: config.Org, Repo: config.Repo}
+	for i, orgRepo := range []string{
+		path.Join(config.Org, config.Repo),
+		path.Join("openshift", "cluster-version-operator"),
+		path.Join("openshift", "ci-tools"),
+	} {
+		src := path.Join(configDir, orgRepo, ciopconfig.SupplementalProwConfigFileName)
+		logrus.WithField("src", src).Info("copying ...")
+		if _, err := os.Stat(src); errors.Is(err, os.ErrNotExist) {
+			continue
+		}
+		if _, err := fileutils.CopyFile(src, path.Join(tmpDir, fmt.Sprintf("%d%s", i, ciopconfig.SupplementalProwConfigFileName))); err != nil {
+			return fmt.Errorf("failed to copy file: %w", err)
+		}
+	}
+
 	agent := prowconfig.Agent{}
-	if err := agent.Start(configPath, "", nil, ""); err != nil {
+	if err := agent.Start(configPath, "", []string{tmpDir}, ciopconfig.SupplementalProwConfigFileName); err != nil {
 		return fmt.Errorf("could not load Prow configuration: %w", err)
 	}
 
 	prowConfig := agent.Config()
-	editProwConfig(prowConfig, config)
 
-	data, err := yaml.Marshal(prowConfig)
-	if err != nil {
-		return fmt.Errorf("could not marshal Prow configuration: %w", err)
-	}
-
-	return ioutil.WriteFile(configPath, data, 0644)
```

Comment:
We write the updated one into the globe one.
Because we never updated anything, we write the same config back to the global config.
Otherwise, we would notice the bug earlier.

The PR writes the config for the new repo into its own folder instead.